api/handlers/refererhandler.py

```diff
@@ -88,6 +96,31 @@ def post(self, cont_name, cid):
         else:
             self.abort(500, 'Analysis not added for container {} {}'.format(cont_name, cid))
 
+    def put(self, cont_name, **kwargs):
+        cid = kwargs.pop('cid')
+        _id = kwargs.pop('_id')
+
+        parent = self.storage.get_parent(cont_name, cid)
+        permchecker = self.get_permchecker(parent)
+        permchecker(noop)('PUT')
+
+
+        payload = self.request.json_body
+        if not payload:
+            self.abort(400, 'PUT request body cannot be empty')
+        self.update_validator(payload, 'PUT')
+
+        payload['modified'] = datetime.datetime.utcnow()
+        
+        try:
+            result = self.storage.update_el(_id, payload)
+        except APIStorageException as e:
```

Comment:
This would only ever get thrown when we can't transform the `_id` to an ObjectId. We have route filters that require an ObjectId structure. I think allowing the APIStorageException to bubble up on it's own and cause a 500 is a better way to handle it, it would show us where there are holes in the filter, if any.